lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -48,7 +48,8 @@ def matches?(given_proc, negative_expectation=false, &block)
           begin
             given_proc.call
           rescue Exception => @actual_error
-            if values_match?(@expected_error, @actual_error)
+            if values_match?(@expected_error, @actual_error) ||
+               (values_match?(@expected_error, @actual_error.message) rescue false)
```

Comment:
i have one test blowing up because of this line (on `spec/rspec/matchers/built_in/raise_error_spec.rb:480`) which i've temporarily dealt with by adding this `rescue` here ( :scream: ). Is this a case that the fuzzy matcher should try to handle? It fails with: `NoMethodError: undefined method `foo' for "FooError":String`